detectors/dna/model.py

```diff
@@ -0,0 +1,41 @@
+from typing import List
+from models.completion import OpenAICompletionModel
+from .config import BlackBoxDnaConfig
+from .ngrams import NGramsProcessor
+from interfaces import IDetector
+import numpy as np
+
+BLACK_BOX_DNA_BYTEARRAY = b"BlackBoxDNADetector"
+
+
+class BlackBoxDNADetector(IDetector):
+    def __init__(self, config: BlackBoxDnaConfig):
+        self.config = config
+        self.model = OpenAICompletionModel(self.config.model_config)
+        self.ngrams_processor = NGramsProcessor(self.config.ngrams_config)
+
+    def get_labels(self) -> List[str]:
+        return ["Human", "AI"]
+
+    def calculate_bscore(self, text: str):
+        words = text.split()
+        divider_num = int(len(words) * (1 - self.config.truncation))
+        beginning, X_ending = " ".join(words[:divider_num]), " ".join(words[divider_num:])
+        Y_endings = self.model.complete_texts([beginning] * self.config.K, False, n_words=[len(words)] * self.config.K)
+        metric = self.ngrams_processor.calculate_metric(X_ending, Y_endings)
+        return metric
+
+    def predict_proba(self, text: str) -> np.array:
+        metric = self.calculate_bscore(text)
+        if metric > self.config.threshold:
+            return np.array([0.0, 1.0])
```

Comment:
sad that this is not probability, need to think of maybe adding generic sigmoid from any scorer as wrapper for such models. Like, you return score, and we have wrapper that returns something like $\frac{1}{1+exp(k \cdot score)}$ and fits k with some cross-validation